tests/spaces/test_dict.py

```diff
@@ -25,18 +27,18 @@ def test_dict_init():
     ):
         Dict(a=Discrete(2), b="Box")
 
-    with pytest.warns(None) as warnings:
+    with warnings.catch_warnings(record=True) as caught_warnings:
```

Comment:
Shouldn't we keep doing testing stuff with `pytest`? Is there a meaningful difference in functionality?

Reply:
The `pytest.warn(None)` is deprecated in pytest 7
This is the new recommended style